### PR TITLE
merge optimizations

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
# benchmark

**Comment by bheni** [comment]
#benchmwark

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

File: go/store/types/tuple.go [hunk 1]
```diff
@@ -268,9 +268,67 @@ func walkTuple(nbf *NomsBinFormat, r *refWalker, cb RefCallback) error {
...
+
+	w := binaryNomsWriter{buff: tf.buffer[tf.pos:], offset: 0}
+	t, err := newTuple(tf.nbf, w, values...)
+
+	if err != nil {
+		return Tuple{}, err
+	}
+
+	n := len(t.buff)
+	t.buff = t.buff[:n]
```
Comment (zachmu) [inline go/store/types/tuple.go:323]: Don't do this brian wtf

File: go/store/types/tuple.go [hunk 2]
```diff
@@ -268,9 +268,79 @@ func walkTuple(nbf *NomsBinFormat, r *refWalker, cb RefCallback) error {
...
+	blockSize := initialBufferSize * approxCapacity
+	return &TupleFactory{
+		buffer:         make([]byte, blockSize),
+		biggestTuple:   initialBufferSize,
+		approxCapacity: approxCapacity,
+	}
+}
+
+// Reset is called when a TupleFactory is reused in a new context as you might want when pooling these.  Reset does
+// not reuse the buffer as the memory may be in use by tuples that have not been collected and reference the same
```
Comment (zachmu) [inline go/store/types/tuple.go:294]: not reset the buffer?

**Review by zachmu — APPROVED** [review]
LGTM

File: go/store/types/tuple.go (hunk 1 above)
Comment (timsehn) [reply go/store/types/tuple.go:323]: We're not writing COD6 here!

Comment (bheni) [reply go/store/types/tuple.go:323]: You can't stop me! "Call of Dolts 6" is gonna be epic!

**Pull request merged**